### PR TITLE
ROX-23709: Disable RHSSO issuer for fleetshard authZ

### PR DESCRIPTION
**Pull request by kovayur**
## Description
Leave only service account authZ for the private api (fleetshard-sync) in fleet manager

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup
make verify lint binary test test/integration
```

### CODE REVIEW

File: dev/env/scripts/lib.sh [hunk 1]
```diff
@@ -352,7 +352,7 @@ is_openshift_cluster() {
 }
 
 delete_tenant_namespaces() {
-    central_namespaces=$($KUBECTL get namespace -o jsonpath='{range .items[?(@.status.phase == "Active")]}{.metadata.name}{"\n"}{end}' | grep '^rhacs-.*$' || true)
+    central_namespaces=$($KUBECTL get namespace -o jsonpath='{range .items[?(@.status.phase == "Active")]}{.metadata.name}{"\n"}{end}' | grep -E '^rhacs-[0-9a-v]{20}$' || true)
```
Comment (kovayur) [inline dev/env/scripts/lib.sh:355]: keep rhacs-vertical-pod-autoscaler

Comment (johannes94) [reply dev/env/scripts/lib.sh:355]: Consider adding a comment that the regex is based on https://github.com/rs/xid can help debug in the future in the unexpected case that this changes.

**Review by johannes94 — COMMENTED** [review]
I don't fully understand how the local dev authentication is now working. Previously I could authenticate to FM with my OCM token.

Now it's only K8s service account.

Lets say I run FM on my local machine, previously I could:

```
make db/teardown db/setup db/migrate
make binary
./fleet-manager serve --dataplane-cluster-config-file dev/config/dataplane-cluster-configuration-infractl-osd.y
```

And then call the API with my OCM token. How would I now configure FM properly, especially intrested in the case the cluster I'm running against doesn't publicly expose it's JWKS.

**Comment by kovayur** [comment]
> I don't fully understand how the local dev authentication is now working. Previously I could authenticate to FM with my OCM token.

This change affects the **private** API only. Public and admin API remain unchanged. You still create centrals with an OCM token. A discovery mechanism has been implemented to connect fleetshard-sync to fleet-manager running locally. Fleet-manager loads the OpenID cluster configuration using the KUBECONFIG file.

File: dev/env/scripts/lib.sh (hunk 1 above)
Comment (kovayur) [reply dev/env/scripts/lib.sh:356]: Done.

File: dev/env/scripts/lib.sh [hunk 2]
```diff
@@ -363,7 +363,7 @@ delete_tenant_namespaces() {
     done
     log "Waiting for leftover RHACS namespaces to be deleted... "
     while true; do
-        central_namespaces=$($KUBECTL get namespace -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' | grep '^rhacs-.*$' || true)
+        central_namespaces=$($KUBECTL get namespace -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' | grep -E '^rhacs-[0-9a-v]{20}$' || true)
```
[the remark this answers is not in the document]
Comment (kovayur) [reply dev/env/scripts/lib.sh:367]: Done.

File: dev/config/gitops-config.yaml [hunk 3]
```diff
@@ -8,17 +8,7 @@ rhacsOperators:
...
-      name: vpa-1
-      imagePullSecrets: [{ name: redhat-pull-secret }]
-      recommendationMarginFraction: 0.30
-      podRecommendationMinCpuMillicores: 10
-    - image: registry.redhat.io/openshift4/ose-vertical-pod-autoscaler-rhel9@sha256:d268de83bd371a725c0d0dad8a48008afa982c10e2997b24ff6d139d90d964f6
-      name: vpa-2
-      imagePullSecrets: [{ name: redhat-pull-secret }]
-      recommendationMarginFraction: 0.30
-      podRecommendationMinCpuMillicores: 20
+  recommenders: []
```
[the remark this answers is not in the document]
Comment (kovayur) [reply dev/config/gitops-config.yaml:11]: Yes, it's unrelated.
I had trouble testing my changes on a local colima cluster and @ludydoo advised me to remove VPAs from the dev gitops config.

```
Error: container has runAsNonRoot and image will run as root (pod: "vpa-1-687c9f87d7-8rrzh_rhacs-vertical-pod-autoscale │
│ r(0ac0e9a9-f0fd-468c-84b5-fe7ca6baa097)", container: recommender)
```

File: e2e/e2e_test.go [hunk 4]
```diff
@@ -561,3 +566,34 @@ func assertEqualSecrets(actualSecrets, expectedSecrets map[string]*corev1.Secret
...
+			Namespace: namespace,
+			Name:      "e2e-tests",
+		},
+	}
+	Expect(ensureTestServiceAccountExists(context.Background(), sa)).To(Succeed())
+
+	token := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			Audiences:         []string{"acs-fleet-manager-private-api"},
+			ExpirationSeconds: pointer.Int64(3600),
```
[the remark this answers is not in the document]
Comment (kovayur) [reply e2e/e2e_test.go]: Changed to 6 hours.

**Review by johannes94 — APPROVED** [review]

**Pull request merged**